lightblue-client-integration-test/src/test/java/com/redhat/lightblue/client/integration/test/LockingTest.java

```diff
@@ -32,13 +36,43 @@ public void testAcquireAndRelease() throws Exception {
         assertTrue(lbLocker.release("test-lock"));
     }
 
+    @Test(expected = LightblueResponseException.class)
+    public void testPingWithError() throws Exception {
+        Locking lbLocker = getLightblueClient().getLocking(LOCK_DOMAIN);
+
+        lbLocker.ping("test-lock");
+    }
+
+    @Test(expected = LightblueResponseException.class)
+    public void testCountWithError() throws Exception {
+        Locking lbLocker = getLightblueClient().getLocking(LOCK_DOMAIN);
+
+        lbLocker.getLockCount("test-lock");
+    }
+
+    @Test(expected = LightblueResponseException.class)
+    public void testReleaseWithError() throws Exception {
+        Locking lbLocker = getLightblueClient().getLocking(LOCK_DOMAIN);
+
+        lbLocker.release("test-lock");
+    }
+
     @Test
     public void testLock() throws LightblueException, IOException {
         Locking lbLocker = getLightblueClient().getLocking(LOCK_DOMAIN);
         try (Lock lock = lbLocker.lock("test-lock")) {
             assertEquals(1, lock.getLockCount());
         }
-        assertEquals(0, lbLocker.getLockCount("test-lock"));
+
+        //assert the lock was successfully released.
+        try {
+            lbLocker.getLockCount("test-lock");
+            fail();
+        } catch (LightblueResponseException e) {
+            Error error = e.getLightblueResponse().getLightblueErrors()[0];
+            assertNotNull(error);
+            assertTrue(error.getMsg().contains("InvalidLockException: test-lock"));
+        }
```

Comment:
I don't see it that way, but that's just semantics. For sake of clarity, I
think lock exceptions can be thought completely separate from response
errors. That makes client code much cleaner. I don't see any benefit in
extending a common exception for this case. Lock errors don't contain any
diagnostic information.

On Wed, Jan 13, 2016 at 12:46 PM, Dennis Crissman <notifications@github.com>
wrote:

> In
> lightblue-client-integration-test/src/test/java/com/redhat/lightblue/client/integration/test/LockingTest.java
> <https://github.com/lightblue-platform/lightblue-client/pull/234#discussion_r49640576>
> :
>
> >      @Test
> >      public void testLock() throws LightblueException, IOException {
> >          Locking lbLocker = getLightblueClient().getLocking(LOCK_DOMAIN);
> >          try (Lock lock = lbLocker.lock("test-lock")) {
> >              assertEquals(1, lock.getLockCount());
> >          }
> > -        assertEquals(0, lbLocker.getLockCount("test-lock"));
> > +
> > +        //assert the lock was successfully released.
> > +        try {
> > +            lbLocker.getLockCount("test-lock");
> > +            fail();
> > +        } catch (LightblueResponseException e) {
> > +            Error error = e.getLightblueResponse().getLightblueErrors()[0];
> > +            assertNotNull(error);
> > +            assertTrue(error.getMsg().contains("InvalidLockException: test-lock"));
> > +        }
>
> I think the argument could be made that a lock exception is a response
> exception. It is just a error on a response from lightblue that happens to
> be for a lock.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-client/pull/234/files#r49640576>
> .
>